### PR TITLE
Flows: Integrate SQLX -- split crates

### PR DESCRIPTION
**Pull request by s373r**
## Description

This PR does the prep work that creates a large diff (in terms of number of files), to reduce reading complexity in following PRs.

Related issue: https://github.com/kamu-data/kamu-cli/issues/578

## Checklist before requesting a review

- [x] [CHANGELOG.md](./CHANGELOG.md) updated
- [x] API changes are backwards-compatible
- [ ] Workspace layout changes include a migration -- not actual
- [ ] Documentation update PR: <link or N/A> -- not actual
- [ ] Dataset pipelines update scheduled if needed -- not actual
- [ ] Unit-tests added -- not actual

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -104,6 +105,15 @@ sqlx-local-clean-sqlite:
 sqlx-prepare:
 	$(foreach crate,$(ALL_DATABASE_CRATES),(cd $(crate) && cargo sqlx prepare );)
 
+###############################################################################
+# Sqlx: add migration
+###############################################################################
+
+.PHONY: sqlx-add-migration
+sqlx-add-migration:
```
Comment (zaychenko-sergei) [inline Makefile:113]: Why would we need this separately from `sqlx-local-setup` ?

File: src/domain/flow-system/services/src/flow/flow_service_inmem.rs
Comment (zaychenko-sergei) [inline src/domain/flow-system/services/src/flow/flow_service_inmem.rs:1]: File and struct were not renamed - should be "impl", not "inmemory"

File: src/infra/flow-system/inmem/tests/tests/test_flow_service_inmem.rs
Comment (zaychenko-sergei) [inline src/infra/flow-system/inmem/tests/tests/test_flow_service_inmem.rs:1]: Flow service tests are supposed to move to services crate

File: src/infra/flow-system/inmem/tests/tests/test_flow_configuration_service_inmem.rs
Comment (zaychenko-sergei) [inline src/infra/flow-system/inmem/tests/tests/test_flow_configuration_service_inmem.rs:1]: Flow configuration service tests should be moved to services crate

File: src/infra/flow-system/inmem/Cargo.toml [hunk 2]
```diff
@@ -27,6 +27,7 @@ kamu-core = { workspace = true }
 kamu-task-system = { workspace = true }
 kamu-task-system-services = { workspace = true }
 kamu-flow-system = { workspace = true }
+kamu-flow-system-services = { workspace = true }
```
Comment (zaychenko-sergei) [inline src/infra/flow-system/inmem/Cargo.toml:30]: There should not be this dependency.
Repositories should be unaware of service layer existence.
Service layer may not link particular repository implementations as well, with the exception of importing this in-memory repository only as dev-dependency for tests.

File: src/infra/flow-system/inmem/tests/tests/utils/manual_flow_trigger_driver.rs
Comment (zaychenko-sergei) [inline src/infra/flow-system/inmem/tests/tests/utils/manual_flow_trigger_driver.rs:1]: These test utils should become part of service test utils. Repository does not use them.

File: Makefile (hunk 1 above)
Comment (s373r) [reply Makefile:113]: It is convenient to have a separate command that will add migrations the moment we need to create them

File: src/infra/flow-system/inmem/Cargo.toml (hunk 2 above)
Comment (s373r) [reply src/infra/flow-system/inmem/Cargo.toml:30]: Eagle eyes -- Done

File: src/infra/flow-system/inmem/tests/tests/test_flow_service_inmem.rs
Comment (s373r) [reply src/infra/flow-system/inmem/tests/tests/test_flow_service_inmem.rs:1]: Done

File: src/infra/flow-system/inmem/tests/tests/utils/manual_flow_trigger_driver.rs
Comment (s373r) [reply src/infra/flow-system/inmem/tests/tests/utils/manual_flow_trigger_driver.rs:1]: Done

File: src/domain/flow-system/services/src/flow/flow_service_inmem.rs
Comment (s373r) [reply src/domain/flow-system/services/src/flow/flow_service_inmem.rs:1]: Thanks! Done (+ files tests)

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**